### PR TITLE
Fix FromJSON1 instance of Sum

### LINKED ISSUE #860: Very minor oopsie in FromJSON1 instance of Sum
In the `InR` case, the `Key` is still `inl`.

```
instance (FromJSON1 f, FromJSON1 g) => FromJSON1 (Sum f g) where
    liftParseJSON p pl (Object (H.toList -> [(key, value)]))
        | key == inl = InL <$> liftParseJSON p pl value <?> Key inl
        | key == inr = InR <$> liftParseJSON p pl value <?> Key inl
...
```

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes  #860

### CODE REVIEW

**Review by phadej — CHANGES_REQUESTED** [review]
Please add test(s) to SerializationFormatSpec.hs

**Comment by Vlix** [comment]
The `... <?> Key x` part of the `Parser` doesn't influence serialization, right?
So the test should be a failed parsing, and make sure the error message is correct?

**Comment by bergmark** [comment]
> The `... <?> Key x` part of the `Parser` doesn't influence serialization, right?
> So the test should be a failed parsing, and make sure the error message is correct?

Right. I think the right place for the test is the golden tests in tests/ErrorMessages.hs (`simple` / `output`)

File: src/Data/Aeson/Types/FromJSON.hs [hunk 1]
```diff
@@ -1801,7 +1801,7 @@ instance (FromJSON1 f, FromJSON1 g, FromJSON a) => FromJSON (Product f g a) wher
 instance (FromJSON1 f, FromJSON1 g) => FromJSON1 (Sum f g) where
     liftParseJSON p pl (Object (H.toList -> [(key, value)]))
         | key == inl = InL <$> liftParseJSON p pl value <?> Key inl
-        | key == inr = InR <$> liftParseJSON p pl value <?> Key inl
+        | key == inr = InR <$> liftParseJSON p pl value <?> Key inr
```
Comment (taasan) [inline src/Data/Aeson/Types/FromJSON.hs:1804]: Wouldn't something like this be less error prone? (Change `f` and `g` to more meaningful names)

```haskell
    liftParseJSON p pl (Object (H.toList -> [(key, value)]))
        | key == inl = f InL
        | key == inr = f InR
      where
        f g = g <$> liftParseJSON p pl value <?> Key key
```

**Comment by phadej** [comment]
I'm sorry. I completely misunderstood the PR. if it's fixing the `Key` only we don't necessarily need a test.

**Pull request merged**